### DISCUSSION TITLE
Access to query in a keybind handler of fzf_exec?

### DISCUSSION THREAD

**Discussion by pkazmier** [ibhagwan/fzf-lua#1569] [Q&A] [answered]
I’m trying to write a keybind handler for fzf_exec that needs access to the query. How would I do that? Use case is for a command I’ve written in both mini.pick and telescope where if the query does not yield results, I invoke a keybind to create a new something based on the query typed.

**Comment by ibhagwan** [comment, marked as answer]
```lua
require'fzf-lua'.fzf_exec("ls", {
  actions = {
    ['enter'] = function(selected, opts)
      print("selected item:", selected[1], 'query', opts.last_query)
    end
  }
})
```

**Reply by ibhagwan** [reply]
If you want to have something a bit more complex like reloading the interface after adding an item, take a look at git branch add action:
https://github.com/ibhagwan/fzf-lua/blob/8330321fa135f7fa9e4855099026c4ad622a0c85/lua/fzf-lua/actions.lua#L610-L628

**Reply by pkazmier** [reply]
This is exactly what I was looking for, but couldn’t find it. Thank you!

**Reply by ibhagwan** [reply]
> This is exactly what I was looking for, but couldn’t find it. Thank you!

YW, the docs are always tricky and there’s always room for improvement :)
